features/step_definitions/integration_steps.rb

```diff
@@ -5,41 +5,67 @@
-  store_secret_in_remote_conjur("#{ENV['PCF_CONJUR_ACCOUNT']}:variable:app/database/password", ci_secret_pass)
+  store_secret_in_remote_conjur("#{ENV['PCF_CONJUR_ACCOUNT']}:variable:app/secrets/org", ci_secret_org)
+  store_secret_in_remote_conjur("#{ENV['PCF_CONJUR_ACCOUNT']}:variable:app/secrets/space", ci_secret_space)
+  store_secret_in_remote_conjur("#{ENV['PCF_CONJUR_ACCOUNT']}:variable:app/secrets/app", ci_secret_app)
 end
 
-Given(/^I install the service broker$/) do
-  install_service_broker
+Given(/^I install the service broker( with preserve policy)?$/) do |preserve|
+  install_service_broker(preserve.present?)
 end
 
 Given(/^I create a service instance for Conjur$/) do
   `cf create-service cyberark-conjur community conjur`
 end
 
-Given(/^I load policy to define my org and space$/) do
-  load_space_policy_in_remote_conjur('ci', 'conjur-service-broker')
-end
-
 When(/^I push the sample app to PCF$/) do
   `cf delete hello-world -f`
   Dir.chdir(integration_test_app_dir) do
     `cf push --no-start --random-route`
   end
 end
 
-When(/^I privilege the app to access the secret in Conjur$/) do
-  entitle_host_in_remote_conjur(app_bind_id)
-end
-
 When(/^I start the app$/) do
   Dir.chdir(integration_test_app_dir) do
     `cf start hello-world`
   end
 end
 
-Then(/^I can retrieve the secret value from the app$/) do
+Then(/^I can retrieve the secret values from the app$/) do
   page_content = ci_app_content
-  expect(page_content).to match(/Database Username: #{ci_secret_user}/)
-  expect(page_content).to match(/Database Password: #{ci_secret_pass}/)
+  expect(page_content).to match(/Org Secret: #{ci_secret_org}/)
+  expect(page_content).to match(/Space Secret: #{ci_secret_space}/)
+  expect(page_content).to match(/App Secret: #{ci_secret_app}/)
+end
+
+Then(/^the policy for the org and space( doesn't)? exist(?:s)?$/) do |negative|
+  expect(remote_conjur_resource_exists?(org_policy_id)).to eq(negative.blank?)
+  expect(remote_conjur_resource_exists?(space_policy_id)).to eq(negative.blank?)
+end
+
+Then(/^the instance resource( doesn't)? exist(?:s)?$/) do |negative|
+  expect(remote_conjur_resource_exists?(instance_resource_id)).to eq(negative.blank?)
+end
+
+When(/^I privilege the org layer to access a secret in Conjur$/) do
+  role = "!layer pcf/ci/#{org_guid('ci')}"
+  secret = "!variable app/secrets/org"
+  privilege_in_remote_conjur(role, secret)
+end
+
+When(/^I privilege the space layer to access a secret in Conjur$/) do
+  role = "!layer pcf/ci/#{org_guid('ci')}/#{space_guid('ci', 'conjur-service-broker')}"
+  secret = "!variable app/secrets/space"
+  privilege_in_remote_conjur(role, secret)
+end
+
+When(/^I privilege the app host to access a secret in Conjur$/) do
+  role = "!host pcf/#{app_bind_id}"
+  secret = "!variable app/secrets/app"
+  privilege_in_remote_conjur(role, secret)
+end
+
+When(/^I remove the service broker$/) do
+  `cf unbind-service hello-world conjur`
+  `cf delete-service conjur -f`
```

Comment:
I don't think this step is needed - it would be uncommon for the service listing to be deleted, I would think

Reply:
With the correct step name, this step is necessary to test the deprovision, right? Or is there something I'm missing?

Reply:
let's call that a brain fart and mark this as resolved 😅